### PR TITLE
Rails 5.1

### PR DESCRIPTION
**Pull request by jfly**
**Not to be merged until after PR #1781**

Upgrading from Rails 5.0 to 5.1 =) I followed the guide [here](http://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-5-0-to-rails-5-1), and the release notes [here](http://guides.rubyonrails.org/5_1_release_notes.html).

Things mostly went pretty smoothly here, there was an issue with our DelayedJob plugin test that I had to fix, and then I addressed all the deprecation warnings due to https://github.com/rails/rails/pull/25337.

Rails 5.1 comes with [system tests](http://guides.rubyonrails.org/5_1_release_notes.html#system-tests), which sound like they could let us get rid of the `database_cleaner` gem. However, rspec has not yet been upgraded to support these system tests (they have a pending PR [here](https://github.com/rspec/rspec-rails/pull/1813)). I've created #1782 to track us fixing this in a later PR.

### CODE REVIEW

File: WcaOnRails/app/uploaders/avatar_uploader_base.rb [hunk 1]
```diff
@@ -62,7 +62,7 @@ def filename
       # This is pretty gross. We only want to reuse the existing filename if
       # a new avatar isn't being uploaded, we look at the *_change attribute to
```
Comment (jonatanklosko) [inline WcaOnRails/app/uploaders/avatar_uploader_base.rb]: The comment needs to be updated as well =)

File: WcaOnRails/app/models/user.rb [hunk 2]
```diff
@@ -315,10 +315,10 @@ def avatar_requires_wca_id
 
   after_save :remove_pending_wca_id_claims
   private def remove_pending_wca_id_claims
-    if delegate_status_changed? && !delegate_status
+    if delegate_status_before_last_save && !delegate_status
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/user.rb]: Didn't you mean `saved_change_to_delegate_status?` here?

File: WcaOnRails/app/uploaders/avatar_uploader_base.rb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/uploaders/avatar_uploader_base.rb]: Whoops, good catch!

File: WcaOnRails/app/models/user.rb (hunk 2 above)
Comment (jfly) [inline WcaOnRails/app/models/user.rb]: Ah, that works too, and is closer to what we used to have. Either way the tests pass. In English, the two sentences are:

1. Did we just change this guy's delegate status and is he currently not a delegate?
2. Was this guy previously a delegate and is he currently not a delegate?

I've changed it from 2) to 1), as you suggested.

**Comment by jfly** [comment]
No objections here, and I'd rather not let this PR sit gathering dust, so I'm going to merge this up!

**Pull request merged**